src/mpp/Dec.hpp

```diff
@@ -1167,6 +1189,37 @@ bool jump_read_optional(BUF& buf, T... t)
 	}
 }
 
+template <size_t I, compact::Family FAMILY, size_t SUBRULE,
+	  class PATH, class BUF, class... T>
+bool jump_read_variant_impl(BUF& buf, T... t)
+{
+	auto&& variant = unwrap(path_resolve(PATH{}, t...));
+	using variant_t = std::remove_reference_t<decltype(variant)>;
+	constexpr size_t size = std::variant_size_v<variant_t>;
+	static_assert(tnt::is_variant_v<variant_t>);
+	static_assert(I < size);
+	using curType = std::variant_alternative_t<I, variant_t>;
+	constexpr auto curFamilies = detectFamily<BUF, curType>();
+
+	if constexpr (family_sequence_contains<FAMILY>(curFamilies)) {
+		variant.template emplace<I>();
+		using NEXT_PATH = path_push_t<PATH, PIT_VARIANT, size, I>;
+		return decode_impl<NEXT_PATH>(buf, t...);
+	} else {
+		return jump_read_variant_impl<I + 1, FAMILY, SUBRULE, PATH>(buf, t...);
```

Comment:
Shouldn't `variant::emplace` encapsulate this matching? See overload (1) from [std::variant](https://en.cppreference.com/w/cpp/utility/variant/emplace).

Reply:
Do you mean matching from `mpp::FAMILY` to variant alternative?

Reply:
Yes. Can't we make it work like `variant.template emplace<FAMILY>()`?

Reply:
How? `std::variant` knows nothing about our msgpack type families.